core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -143,6 +143,10 @@ protected Table newTable(TableOperations ops, String tableName) {
     return new BaseTable(ops, tableName);
   }
 
+  public Table underlyingTable() {
```

Comment:
Question: What about capturing the format version as a field in `SerializableTable`, similar to what we do for the metadata file location? The problem right now is that calling `lazyTable()` may actually require a request to load the metadata, which is something we would want to ideally avoid.